src/main/java/de/zalando/aruha/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -50,6 +50,9 @@ public void onEnter() {
     }
 
     private void onTopologyChanged() {
+        if (topologyListener == null) {
+            throw new IllegalStateException("topologyListener should not be null when calling onTopologyChanged method");
```

Comment:
The usecase for `topologyListener` to be null can be that `onTopologyChanged()` is executed after `onExit()` was called for this state.
Also, I don't see any way how it can fail during listener creation.